### PR TITLE
Add machine readable --output options to `kubectl api-resources`

### LINKED ISSUE #1382: Add machine readable --output options to `kubectl api-resources`
### What would you like to be added?

Currently the `kubectl api-resources` command `--output` flag only takes `wide` or `name`, a machine readable format such as `json` would be useful for scripting/utilizing `kubectl` 

### Why is this needed?

I am a dev at [fig](https://github.com/withfig) and help maintain the [autocomplete repo](https://github.com/withfig/autocomplete), we generate completions for the `kubectl` CLI based on `kubectl api-resources --output name` right now, but this lacks support for short names which are hard to parse from the human readable `kubectl api-resources`, which would be easily parsed from a `kubectl api-resources --output json` command.

This might be a much less common use case for others but it still feels useful to have machine readable output for as many `--output` flags as possible.

### PR DESCRIPTION
**Pull request by XDRAGON2002**
#### What type of PR is this?

/kind feature

#### What this PR does / why we need it:

This PR adds support for json output format for `kubectl api-resources`.

#### Which issue(s) this PR fixes:

Fixes https://github.com/kubernetes/kubectl/issues/1382

#### Does this PR introduce a user-facing change?

```release-note
Adds json as a possible output format. Made changes to apiresources.go which added support for kubectl api-resources --output json
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: staging/src/k8s.io/kubectl/pkg/cmd/apiresources/apiresources.go [hunk 1]
```diff
@@ -235,6 +238,15 @@ func (o *APIResourceOptions) RunAPIResources() error {
 				strings.Join(r.APIResource.Categories, ",")); err != nil {
 				errs = append(errs, err)
 			}
+		case "json":
```
Comment (ardaguclu) [inline staging/src/k8s.io/kubectl/pkg/cmd/apiresources/apiresources.go:241]: Could you please check how other commands handle json printing(i.e. commands support `--output`). We need to also support yaml and that's why, we need to use JsonYamlPrinter.

Since this version of this patch looks hacky;

/hold

Comment (ardaguclu) [reply staging/src/k8s.io/kubectl/pkg/cmd/apiresources/apiresources.go:241]: @XDRAGON2002  It is worth to check how `wait` command handles this https://github.com/kubernetes/kubernetes/blob/369534c6ec625ed9b251b453215c9578cdeb59bc/staging/src/k8s.io/kubectl/pkg/cmd/wait/wait.go#L152

**Comment by XDRAGON2002** [comment]
@ardaguclu Thanks for the quick review! 
I've updated the previous code to use `json.marshal` in order to generate a json output instead.

Though while going through the `printers` approach requested by you, I went through the entire codebase for the printers and ended up [here](https://github.com/kubernetes/kubernetes/blob/369534c6ec625ed9b251b453215c9578cdeb59bc/staging/src/k8s.io/cli-runtime/pkg/printers/json.go#L34).
I tried to directly use this in order to get the desired output by creating a `JSONPrinter` and calling `PrintObj` on it, but since it requires a `runtime.Object` it didn't work with the `groupResource` struct.

Any pointers or suggestions regarding how can I improve the implementation?
